content/en/docs/tasks/manage-kubernetes-objects/declarative-config.md

```diff
@@ -371,44 +371,82 @@ to result in the user deleting something unintentionally:
-You must be careful when using this command, so that you
-do not delete objects unintentionally.
+Only use this if you know what you are doing. You must be careful when using this command, so that you do not delete objects unintentionally. For more information on the problems with this command, see the [Current Implementation](git.k8s.io/enhancements/keps/sig-cli/3659-kubectl-apply-prune#current-implementation) section of the ApplySet KEP.
 {{< /warning >}}
 
-As an alternative to `kubectl delete`, you can use `kubectl apply` to identify objects to be deleted after their
-configuration files have been removed from the directory. Apply with `--prune`
-queries the API server for all objects matching a set of labels, and attempts
-to match the returned live object configurations against the object
+To use allowlist-based pruning, add the following flags to your `kubectl apply` invocation:
+- `--prune`: Delete previously applied objects that are not in the set passed to the current invocation.
+- `--prune-allowlist`: A list of group-version-kinds (GVKs) to consider for pruning. This flag is optional but strongly encouraged, as its [default value](https://git.k8s.io/kubectl/pkg/util/prune/prune.go#L28-L50) is a partial list of both namespaced and cluster-scoped types, which can lead to surprising results.
+- `--selector/-l`: Use a label selector to constrain the set of objects selected for pruning. This flag is optional but strongly encouraged.
+- `--all`: use instead of `--selector/-l` to explicitly select all previously applied objects of the allowlisted types.
+
+Allowlist-based pruning queries the API server for all objects of the allowlisted GVKs that match the given labels (if any), and attempts to match the returned live object configurations against the object
 configuration files. If an object matches the query, and it does not have a
 configuration file in the directory, and it has a `last-applied-configuration` annotation,
 it is deleted.
 
-{{< comment >}}
-TODO(pwittrock): We need to change the behavior to prevent the user from running apply on subdirectories unintentionally.
-{{< /comment >}}
 
 ```shell
-kubectl apply -f <directory/> --prune -l <labels>
+kubectl apply -f <directory/> --prune -l <labels> --prune-allowlist=<gvk-list>
 ```
 
 {{< warning >}}
 Apply with prune should only be run against the root directory
 containing the object configuration files. Running against sub-directories
-can cause objects to be unintentionally deleted if they are returned
-by the label selector query specified with `-l <labels>` and
-do not appear in the subdirectory.
+can cause objects to be unintentionally deleted if they were previously applied, 
+have the labels given (if any), and do not appear in the subdirectory.
+{{< /warning >}}
+
+{{% /tab %}}
+
+{{% tab name="ApplySet" %}}
+
+{{< feature-state for_k8s_version="v1.27" state="alpha" >}}
+
+{{< warning >}}
+`kubectl apply --prune --applyset` is in alpha, and backwards incompatible
```

Comment:
This warning is why I'd probably not document the labels in the other file, not yet. If we decide to go with a resource, rather than label it'll be confusing and we'll have users relying on alpha feature, which is not plainly obvious in that other file.